data-table/table.js

```diff
@@ -305,6 +306,47 @@ function createRowEntryBox(config) {
     createRowDeleteBtn(config);
 }
 
+/**
+ * This function enables the user to enter the name of a column. Creates a field for text input
+ * as well as a button that sends input text to addSingleColumn().
+ * @param {object} config   - Table configuration object
+ * @returns {undefined}     - Doesn't return anything
+ */
+function createColumnInputAndBtn(config) {
+    let entryBoxDiv = document.getElementById(config.tableIds.entryBoxDivId);
+    // Creates the field for text input
+    let input = document.createElement("INPUT");
+    input.type = 'text';
+    input.placeholder = "Enter " + config.datumConfig.names;
+
+    // If the user hits enter while in the text box, click the addColumnBtn
+    input.addEventListener("keyup", function(event) {
+        event.preventDefault();
+
+        if (event.code === "Enter") {
+            addColumnBtn.click();
+        }
+    })
+    entryBoxDiv.appendChild(input);
+
+    // Creates the button that will take the user input and send it to addSingleColumn() when clicked
+    /**
+     * TODO: Button will need to accept a number of columns from the user then pass that number to addMultipleColumns
+     */
+    let addColumnBtn = document.createElement("button");
+    addColumnBtn.innerHTML = "Add column to right";
+    addColumnBtn.onclick = function () {
+        let value = input.value.trim();
+        if (value !== '') {
+            addSingleColumn(config, config.numColumns, value);
+        } else {
+            addSingleColumn(config, config.numColumns, config.datumConfig.names[0])
```

Comment:
Similarly, I believe you want `config.columnsName` here.

It looks like we are assuming here that both the rows and the columns are modifiable. @khashshafiei can you add a story to make sure we handle this configuration option? Specifically, whether rows or columns have modifiable titles should be configurable.

For example, in the RCVis use case: rows (candidate names) is configurable, but columns (rounds) is not.